plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoPageSource.java

```diff
@@ -149,7 +151,12 @@ public Page getNextPage()
             pageBuilder.declarePosition();
             for (int column = 0; column < columnTypes.size(); column++) {
                 BlockBuilder output = pageBuilder.getBlockBuilder(column);
-                appendTo(columnTypes.get(column), currentDoc.get(columnNames.get(column)), output);
+                if (this.type == MongoTableHandle.Type.QUERY) {
+                    columnTypes.get(column).writeSlice(output, jsonParse(utf8Slice(toVarcharValue(currentDoc))));
```

Comment:
Does this mean that we support only `VARCHAR` columns if the table type is `QUERY` ?

Reply:
Can we allow the user to specify a column like they do for normal tables ?